contracts/lib/ExecutionLib.sol

```diff
@@ -17,8 +17,15 @@ library ExecutionLib {
         //bytes calldata data  = userOpCallData;
         assembly {
             let baseOffset := add(userOpCallData.offset, 0x24) //skip 4 bytes of selector and 32 bytes of execution mode
-            erc7579ExecutionCalldata.offset := add(baseOffset, calldataload(baseOffset))
+            let calldataLoadOffset := calldataload(baseOffset)
+            // check for potential overflow in calldataLoadOffset
+            if gt(calldataLoadOffset, 0xffffffffffffffff) { revert(0, 0) }
+            erc7579ExecutionCalldata.offset := add(baseOffset, calldataLoadOffset)
             erc7579ExecutionCalldata.length := calldataload(sub(erc7579ExecutionCalldata.offset, 0x20))
+            // TODO: check is sufficient
+            // forgefmt: disable-next-item
+            if lt(add(erc7579ExecutionCalldata.offset, erc7579ExecutionCalldata.length), 
```

Comment:
I'm not sure if this is right. Imo we need to check that erc7579ExecutionCalldata is within userOpCallData

so the checks should be:

erc7579ExecutionCalldata.offset < userOpCallData.offset + userOpCallData.length
erc7579ExecutionCalldata.offset + erc7579ExecutionCalldata.length < userOpCallData.offset + userOpCallData.length

not 100% on this tho so pls double check

Reply:
also maybe second condition should be <=